### DISCUSSION TITLE
Is there a diagnostic rule for missing function return type annotation?

### DISCUSSION THREAD

**Discussion by andyluooo** [microsoft/pyright#5077] [Q&A] [answered]
I want pyright to warn me when I forget to declare the function return type, but I failed to find such a rule. Will there be one in the future? Alternatively, I find the pylance inlay hints useful, but I want it not to show up in an `__init__` method. Is there a way to configure this?

**Comment by erictraut** [comment, marked as answer]
Please refer to [this issue](https://github.com/microsoft/pyright/issues/3013) for a discussion about enforcing return type annotations. This has been discussed at length and considered multiple times but ultimately rejected each time.

For suggestions about pylance functionality, please post to the [pylance-release project](https://github.com/microsoft/pylance-release).
